### ISSUE TITLE
[kubernetes] Failed to remove remote node after restarting the flowbroker

### ISSUE THREAD

**Issue by betefaber** [dojot/dojot#1809]
**k8s environment**, baseline **v0.4.4-rc.1**

----
test 1: NOK
----
Steps:
- restart dojot
- add remote node
- **restart flowbroker**
- **restart pod of remote node** (the node does not persisted)

![teste3](https://user-images.githubusercontent.com/37310063/97052931-4db5d680-1558-11eb-829a-d6e1a1870fc9.png)

- delete remote node
```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"message":"Failed to remove node.","error":"Not found remote node with id kelvin."}
```
```
<20:00:33 23/10/2020> -- |api -- ERROR: Error: Not found remote node with id kelvin.
    at Promise (/opt/flowbroker/orchestrator/src/nodeManager.js:349:25)
    at new Promise (<anonymous>)
    at NodeManager.delRemoteNode (/opt/flowbroker/orchestrator/src/nodeManager.js:318:12)
    at app.delete (/opt/flowbroker/orchestrator/src/api.js:78:15)
    at Layer.handle [as handle_request] (/opt/flowbroker/orchestrator/node_modules/express/lib/router/layer.js:95:5)
    at next (/opt/flowbroker/orchestrator/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/opt/flowbroker/orchestrator/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/opt/flowbroker/orchestrator/node_modules/express/lib/router/layer.js:95:5)
    at /opt/flowbroker/orchestrator/node_modules/express/lib/router/index.js:281:22
    at param (/opt/flowbroker/orchestrator/node_modules/express/lib/router/index.js:354:14)
    at param (/opt/flowbroker/orchestrator/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/opt/flowbroker/orchestrator/node_modules/express/lib/router/index.js:410:3)
    at next (/opt/flowbroker/orchestrator/node_modules/express/lib/router/index.js:275:10)
    at serveStatic (/opt/flowbroker/orchestrator/node_modules/serve-static/index.js:75:16)
    at Layer.handle [as handle_request] (/opt/flowbroker/orchestrator/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/opt/flowbroker/orchestrator/node_modules/express/lib/router/index.js:317:13)

```
----
test 2: NOK (#1803)
----
Steps:
- add remote node
- **restart flowbroker**
- delete remote node

```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"message":"Failed to remove node.","error":"Not found remote node with id kelvin."}
```
----
test 3 (happy day): OK
---
Steps:
- restart dojot
- add remote node
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{"message":"ok"}
```
```
<19:06:17 23/10/2020> -- |nodeMngr -- DEBUG: Loading remote nodes for tenant admin ...
<19:06:17 23/10/2020> -- |nodeMngr -- DEBUG: ...Loaded remote nodes for tenant admin.
<19:06:17 23/10/2020> -- |nodeMngr -- DEBUG: Succeeded to set manager to handle processing nodes for tenant admin
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-Y4GkoUaBu_FkbmYiPc5eWg' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- DEBUG: Connections established with RabbitMQ!
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-moQXcydGfCa9h4WlVOY9_Q' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-ImAIP0pFvGyXFKVx-M5_Pg' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-LJyGhhjsal9UwFnap_P6bg' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-J4mlshfJC_kAe9BGav0jkQ' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-D0KRxgymelciJrvjFRxxZQ' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: Producer ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-_UpuRGn62CwsmoxSjqIPJw' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-UXZ8pXGKU53OnHhzZnNp1A' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-DhJA3q9R6nWuVg58iPTfCA' } ready ...
<19:06:18 23/10/2020> -- |amqp -- INFO: consumer { consumerTag: 'amq.ctag-vS9Rp1Q4oz56OHh1Xu4zPw' } ready ...
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Using kubernetes driver.
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Selected engine: kubernetes
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Using kubernetes API servera @ https://10.96.0.1:443
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Testing access...
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Retrieving current deployment...
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Retrieving current deployment...
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<19:12:23 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Current flowbroker deployments are: { statusCode: 200,
 ...
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... server access is OK.
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Current flowbroker deployments are: { statusCode: 200,
...
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Building deployment creation request...
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Adding container dojot/kelvin-example:3.0.0-alpha2 to the set...
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... container adminkelvin was added to the set.
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment creation request was built.
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Deployment is:
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: { apiVersion: 'apps/v1',
  kind: 'Deployment',
  metadata:
   { labels: { app: 'flowbroker', name: 'flownode-adminkelvin' },
     name: 'flownode-adminkelvin' },
  spec: (96%)
   { selector:
      { matchLabels: { app: 'flowbroker', name: 'flownode-adminkelvin' } },
     replicas: 1,
     template:
      { metadata: { labels: { name: 'flownode-adminkelvin', app: 'flowbroker' } },
        spec:
         { containers:
            [ { name: 'adminkelvin',
                image: 'dojot/kelvin-example:3.0.0-alpha2',
                imagePullPolicy: 'Always',
                ports: [ { name: 'amqp', port: 5555, containerPort: 5555 } ] } ],
           restartPolicy: 'Always' } } } }
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Creating service for this deployment...
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Service to be created: { apiVersion: 'v1',
  kind: 'Service',
  metadata:
   { name: 'flownode-adminkelvin',
     labels: { app: 'flowbroker', name: '' } },
  spec:
   { selector: { name: 'flownode-adminkelvin' },
     ports: [ { protocol: 'TCP', port: 5555, targetPort: 5555 } ] } }
<19:12:24 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... service for deployment created:  { statusCode: 201,
  body:
   { kind: 'Service',
     apiVersion: 'v1',
     metadata:
      { name: 'flownode-adminkelvin',
        namespace: 'dojot',
        selfLink: '/api/v1/namespaces/dojot/services/flownode-adminkelvin',
        uid: '7632f8d1-ed97-4226-b698-397abd2ddbb0',
        resourceVersion: '18207518',
        creationTimestamp: '2020-10-23T19:12:24Z',
        labels: { app: 'flowbroker', name: '' } },
     spec:
      { ports: [ { protocol: 'TCP', port: 5555, targetPort: 5555 } ],
        selector: { name: 'flownode-adminkelvin' },
        clusterIP: '10.96.210.101',
        type: 'ClusterIP',
        sessionAffinity: 'None' },
     status: { loadBalancer: {} } } }
<19:12:24 23/10/2020> -- |dispatcher -- INFO: Dispatcher connected to flownode-adminkelvin on port 5555
<19:12:29 23/10/2020> -- |dispatcher -- DEBUG: Received reply: {"payload":{"payload":{"id":"sample/kelvin","name":"kelvin","module":"kelvin","version
":"0.0.0"}},"requestId":"4d26ea73-95e7-47e3-9ea5-34d28ce07c08"}
<19:12:29 23/10/2020> -- |dispatcher -- DEBUG: Received reply: {"payload":{"payload":"<script type=\"text/x-red\" data-template-name=\"kelvin\">\n
 <div class=\"form-row\">\n        <label for=\"node-input-name\"><i class=\"fa fa-tag\"></i> Name</label>\n        <input type=\"text\" id=\"node-input-name\">\
n    </div>\n    <div class=\"form-row\">\n        <label for=\"node-input-in\"><i class=\"fa fa-edit\"></i> <span>input from</span></label>\n        <input type
=\"text\" id=\"node-input-in\" placeholder=\"payload\" style=\"width:250px;\">\n        <input type=\"hidden\" id=\"node-input-inFieldType\">\n    </div>\n    <d
iv class=\"form-row\">\n        <label for=\"node-input-out\"><i class=\"fa fa-edit\"></i> <span>output to</span></label>\n        <input type=\"text\" id=\"node
-input-out\" placeholder=\"payload\" style=\"width:250px;\">\n        <input type=\"hidden\" id=\"node-input-outFieldType\">\n    </div>\n</script>\n\n<script ty
pe=\"text/x-red\" data-help-name=\"kelvin\">\n</script>\n\n<script type=\"text/javascript\">\n    RED.nodes.registerType('kelvin', {\n        category: 'converte
rs',\n        color: \"rgb(231, 231, 174)\",\n        defaults: {\n            name: { value: \"\", required: false },\n            out:  { value: \"payload\", v
alidate: RED.validators.typedInput(\"outFieldType\") },\n            outFieldType: { value: \"msg\" },\n            in: { value: \"payload\", validate: RED.valid
ators.typedInput(\"inFieldType\") },\n            inFieldType: { value: \"msg\" },\n        },\n        inputs: 1,\n        outputs: 1,\n        icon: \"cog.png\
",\n        label: function () {\n            return this.name || \"to kelvin\";\n        },\n        labelStyle: function () {\n            return this.name ? \
"node_label_italic\" : \"\";\n        },\n        oneditprepare: function () {\n            if (!this.outFieldType) {\n                this.outFieldType = 'msg';
\n            }\n\n            $(\"#node-input-out\").typedInput({\n                default: 'msg',\n                types: ['msg'],\n                typeField:
$(\"#node-input-outFieldType\")\n            });\n\n            if (!this.inFieldType) {\n                this.inFieldType = 'msg';\n            }\n\n
 $(\"#node-input-in\").typedInput({\n                default: 'msg',\n                types: ['msg'],\n                typeField: $(\"#node-input-inFieldType\")\
n            });\n        },\n        oneditsave: function () { }\n    });\n</script>"},"requestId":"19352af6-250c-44ce-82fe-0a5ffb138a03"}
<19:12:29 23/10/2020> -- |nodeMngr -- DEBUG: Succeeded to add remote node with id kelvin
```
```
curl -X GET "${DOJOT_HOST}/flows/v1/node" -H "Authorization: Bearer ${JWT}" -H "Content-Type:application/json"
{"nodes":[{"id":"kelvin","image":"dojot/kelvin-example:3.0.0-alpha2"}]}
```
- delete remote node
```
 curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"message":"ok"}
```
```
<19:23:50 23/10/2020> -- |nodeMngr -- DEBUG: Trying to remove docker container undefined
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Retrieving current deployment...
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Current flowbroker deployments are: { statusCode: 200,
...
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment adminkelvin...
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Removing container adminkelvin from the set.
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Current container list is
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Scaling down deployment flownode-adminkelvin...
<19:23:50 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment scale down request was sent to the server.
<19:23:51 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment flownode-adminkelvin was scaled down.
<19:23:51 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment flownode-adminkelvin...
<19:23:51 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment removal request was sent to the server.
<19:23:51 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment flownode-adminkelvin was removed.
<19:23:51 23/10/2020> -- |kb8sRemoveNode -- DEBUG: Removing service for this deployment...
<19:23:51 23/10/2020> -- |kb8sRemoveNode -- DEBUG: ... Service for deployment removed:  { statusCode: 200,
  body:
   { kind: 'Status',
     apiVersion: 'v1',
     metadata: {},
     status: 'Success',
     details:
      { name: 'flownode-adminkelvin',
        kind: 'services',
        uid: '7632f8d1-ed97-4226-b698-397abd2ddbb0' } } }
<19:23:51 23/10/2020> -- |nodeMngr -- DEBUG: Succeeded to remove container.
<19:23:51 23/10/2020> -- |nodeMngr -- DEBUG: Succeeded to delete remote node with id kelvin for tenant admin.

```
```
 curl -X GET "${DOJOT_HOST}/flows/v1/node" -H "Authorization: Bearer ${JWT}" -H "Content-Type:application/json"
 {"nodes":[]}
```

----
test 4: OK
----
Steps:
- restart dojot
- add remote node
- restart pod of remote node
```
kubectl delete pod/flownode-adminkelvin-77659695-7q48q
```
The node has been persisted and the flow is performed

- delete remote node
```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"message":"ok"}
```

**Comment by betefaber** [comment]
#1365 ?

**Comment by betefaber** [comment]
Fixed in **v0.5.0-rc.2**

----
Test 1: OK
----
- adding remote node
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{
  "message": "ok"
}
```
```
<14:52:25 24/11/2020> -- |nodeMngr -- DEBUG: Succeeded to add remote node with id kelvin
```

- restarting flowbroker

```
kubectl delete pod/flowbroker-548d4b849f-gbt6w
pod "flowbroker-548d4b849f-gbt6w" deleted
```
- restating remote node
```
kubectl delete pod/flownode-adminkelvin-77659695-l6d97
pod "flownode-adminkelvin-77659695-l6d97" deleted
```
- remove remote node
```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{
  "message": "ok"
}
```
```
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment adminkelvin...
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: Removing container adminkelvin from the set.
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: Current container list is
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: Scaling down deployment flownode-adminkelvin...
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment scale down request was sent to the server.
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment flownode-adminkelvin was scaled down.
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment flownode-adminkelvin...
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment removal request was sent to the server.
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment flownode-adminkelvin was removed.
<15:03:11 24/11/2020> -- |kb8sRemoveNode -- DEBUG: Removing service for this deployment...
<15:03:12 24/11/2020> -- |kb8sRemoveNode -- DEBUG: ... Service for deployment removed:  { statusCode: 200,
  body:
   { kind: 'Status',
     apiVersion: 'v1',
     metadata: {},
     status: 'Success',
     details:
      { name: 'flownode-adminkelvin',
        kind: 'services',
        uid: '393f5b44-bec1-4f75-8030-8de3daa16e59' } } }
<15:03:12 24/11/2020> -- |nodeMngr -- DEBUG: Succeeded to remove container.
<15:03:12 24/11/2020> -- |nodeMngr -- DEBUG: Succeeded to delete remote node with id kelvin for tenant admin.
```
----
Test 2: OK
----
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{
  "message": "ok"
}
```
```
kubectl delete pod/flowbroker-548d4b849f-prcjq
pod "flowbroker-548d4b849f-prcjq" deleted
```
```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" 2>/dev/null | jq
{
  "message": "ok"
}
```
----
Test 3: OK
----
```
curl -X GET ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" 2>/dev/null | jq
{
  "nodes": []
}
```
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{
  "message": "ok"
}
```
```
curl -X GET ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" 2>/dev/null | jq
{
  "nodes": [
    {
      "id": "kelvin",
      "image": "dojot/kelvin-example:3.0.0-alpha2"
    }
  ]
}
```
```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" 2>/dev/null | jq
{
  "message": "ok"
}
```

----
Test 4: OK
----
```
curl -X GET ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" 2>/dev/null | jq
{
  "nodes": []
}
```
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{
  "message": "ok"
}
```
```
curl -X GET ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{
  "nodes": [
    {
      "id": "kelvin",
      "image": "dojot/kelvin-example:3.0.0-alpha2"
    }
  ]
}
```
```
kubectl delete pod/flownode-adminkelvin-77659695-j8fk2
pod "flownode-adminkelvin-77659695-j8fk2" deleted
```

```
curl -X DELETE ${DOJOT_HOST}/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" 
{
  "message": "ok"
}
```
```
<15:19:19 24/11/2020> -- |nodeMngr -- DEBUG: Succeeded to delete remote node with id kelvin for tenant admin.
```
